p2p/message.go

```diff
@@ -304,8 +315,17 @@ func (bm *BlockAnnounceMessage) Decode(r io.Reader) error {
 	return err
 }
 
+// Id returns the unique Id of the block
+func (bm *BlockAnnounceMessage) Id() string {
+	hash, err := common.Blake2bHash([]byte(bm.String()))
```

Comment:
I think this returns the block hash, not the ID of the message?  Could you rename the function to be more clear